blues/reporters.py

```diff
@@ -0,0 +1,381 @@
+        self._handle.root._v_attrs.reference = np.string_('DOI: 10.1021/acs.jpcb.7b11820')
+
+        if not hasattr(self._handle.root._v_attrs, 'application'):
+            self._handle.root._v_attrs.application = np.string_('OpenMM')
+            self._handle.root._v_attrs.applicationVersion = np.string_(simtk.openmm.version.full_version)
+
+        # create arrays that store frame level informat
+        if set_coordinates:
+            self._create_earray(where='/', name='coordinates',
+                atom=self.tables.Float32Atom(), shape=(0, self._n_atoms, 3))
+            self._handle.root.coordinates.attrs['units'] = np.string_('nanometers')
+
+        if set_time:
+            self._create_earray(where='/', name='time',
+                atom=self.tables.Float32Atom(), shape=(0,))
+            self._handle.root.time.attrs['units'] = np.string_('picoseconds')
+
+        if set_cell:
+            self._create_earray(where='/', name='cell_lengths',
+                atom=self.tables.Float32Atom(), shape=(0, 3))
+            self._create_earray(where='/', name='cell_angles',
+                atom=self.tables.Float32Atom(), shape=(0, 3))
+            self._handle.root.cell_lengths.attrs['units'] = np.string_('nanometers')
+            self._handle.root.cell_angles.attrs['units'] = np.string_('degrees')
+
+        if set_velocities:
+            self._create_earray(where='/', name='velocities',
+                atom=self.tables.Float32Atom(), shape=(0, self._n_atoms, 3))
+            self._handle.root.velocities.attrs['units'] = np.string_('nanometers/picosecond')
+
+        if set_kineticEnergy:
+            self._create_earray(where='/', name='kineticEnergy',
+                atom=self.tables.Float32Atom(), shape=(0,))
+            self._handle.root.kineticEnergy.attrs['units'] = np.string_('kilojoules_per_mole')
+
+        if set_potentialEnergy:
+            self._create_earray(where='/', name='potentialEnergy',
+                atom=self.tables.Float32Atom(), shape=(0,))
+            self._handle.root.potentialEnergy.attrs['units'] = np.string_('kilojoules_per_mole')
+
+        if set_temperature:
+            self._create_earray(where='/', name='temperature',
+                atom=self.tables.Float32Atom(), shape=(0,))
+            self._handle.root.temperature.attrs['units'] = np.string_('kelvin')
+
+        #Add another portion akin to this if you want to store more data in the h5 file
+        if set_alchemicalLambda:
+            self._create_earray(where='/', name='alchemicalLambda',
+                atom=self.tables.Float32Atom(), shape=(0,))
+            self._handle.root.alchemicalLambda.attrs['units'] = np.string_('dimensionless')
+
+        if set_protocolWork:
+            self._create_earray(where='/', name='protocolWork',
+                atom=self.tables.Float32Atom(), shape=(0,))
+            self._handle.root.protocolWork.attrs['units'] = np.string_('kT')
+
+        if parameters:
```

Comment:
I'm not super familiar with HDF5, so can you clarify if you set up the `parameters`/`environment` writing so that they only are written once (since they won't change through a given simulation)?